### ISSUE TITLE
Use in production?

### ISSUE THREAD

**Issue by lonix1** [dj-nitehawk/FastEndpoints#93]
I'm impressed how this library simplifies a typical API app. It's very clean and sensible. I imagine I'll use it on anything new I work on, especially for personal stuff.

But this is still relatively new - so I'm curious - has anyone used this successfully in production?

I realise speed is it's major selling point, but I'm interested in the day to day stuff (speed is just one feature of many) - how compatible is it with the many libraries one typically uses, pain points, any risks, maintainability, testability, etc.

Basically, any real-world advice: experiences, tips, etc.

Thanks for this cool library!

**Comment by dj-nitehawk** [comment]
the company i work for has a mega modular monolith + a few microservices totalling more than 400 endpoints in production ever since v2 of fastendpoints, and the whole team is having a blast with the dev experience :-)

i've also seen a few (at least 8) closed source projects that are doing well in production. all the open source projects can be found here: https://github.com/dj-nitehawk/FastEndpoints/network/dependents?package_id=UGFja2FnZS0yNzQ2NDI4ODMz

compatibility - is pretty good since it just uses the asp.net pipeline. swagger/serilog/automapper/etc. (automapper is a no-no in my teams though)

risk/maintainability - only time will tell. so far so good ;-)

testability - supports unit tests as well as integration/functional testing

bottom line is: you yourself will have to try it out as the project is still young.

let's see what others have to say.

**Comment by lonix1** [comment]
Hey @dj-nitehawk thanks for the personal reply!

So I'm clear, your library is used by your employer for actual production work? That is very reassuring because unlike other OSS libraries, it means you have incentive to maintain + fix + update it along the way. You should consider mentioning that somewhere, it's a very strong selling point.

I use all those libraries you mentioned, so I'm very happy (I also don't use AutoMapper anymore :-))

I was comparing this library to "Ardalis/Endpoints", which is really nice too, but this one captured my imagination because it's so clean as I mentioned above, without really long generic/lambda code. Thanks for making it!

**Comment by lonix1** [comment]
I'm closing this to keep the issue tracker clean. But if you have production experience / tips / advice, please post below! Thank you...

**Comment by dj-nitehawk** [comment]
my pleasure and yes we have a few systems in production and all our new projects are done with fastendpoints as well, instead of mvc.
